tests/firebird/test-firebird.cpp

```diff
@@ -1297,6 +1161,17 @@ struct TableCreatorCLOB : public tests::table_creator_base
     }
 };
 
+struct TableCreatorBLOB : public tests::table_creator_base
+{
+    TableCreatorBLOB(soci::session & sql)
+            : tests::table_creator_base(sql)
+    {
+        sql << "create table soci_test(id integer, b blob)";
+        sql.commit();
+        sql.begin();
```

Comment:
This is weird as it's done before starting the transaction in the caller, is it really supposed to work like this?

Reply:
I'm not 100% sure why this is needed, but given that I explicitly added it in a commit that claims to be fixing failing Firebird test cases, I would say it probably serves some purposes.
If you look at other table creator implementations for Firebird that use blob types (CLOB and XML) you'll find that they use the same construct (and those existed in this way before I started messing with the code :D)

Reply:
Incidentally those constructs seem to have been originally introduced by yourself - see https://github.com/SOCI/soci/commit/059faf5d961b01f9c816a11a9ac747dd3c9f5b47 and https://github.com/SOCI/soci/commit/9943459f405ade3002f16fbd238fb7acffd4d3c1

Reply:
This is because nobody reviews my PRs :-/

But I now seem to remember that Firebird has some weird transaction semantics, different from all the other databases and implicitly starts one even if you don't do it explicitly. So let's keep this if this is what it needs to work, thanks.